### PR TITLE
fixed issue #170, hopefully

### PR DESCRIPTION
**Pull request by hgdsraj**
Fixed issue #170 
https://github.com/atom/language-c/issues/170

Added virtual keyword to storage.type.modfier's list.

### CODE REVIEW

File: grammars/c++.cson [hunk 1]
```diff
@@ -277,7 +277,7 @@
         ]
       }
       {
-        'begin': '\\b(class|struct)\\b\\s*([_A-Za-z][_A-Za-z0-9]*\\b)?+(\\s*:\\s*(public|protected|private)\\s*([_A-Za-z][_A-Za-z0-9]*\\b)((\\s*,\\s*(public|protected|private)\\s*[_A-Za-z][_A-Za-z0-9]*\\b)*))?'
+        'begin': '\\b(class|struct)\\b\\s*([_A-Za-z][_A-Za-z0-9]*\\b)?+(\\s*:\\s*(public|protected|private|virtual)\\s*([_A-Za-z][_A-Za-z0-9]*\\b)((\\s*,\\s*(public|protected|private)\\s*[_A-Za-z][_A-Za-z0-9]*\\b)*))?'
```
Comment (50Wliu) [inline grammars/c++.cson]: You forgot to add `virtual` to the second time the public|protected|private list occurs.

Comment (hgdsraj) [inline grammars/c++.cson]: Added :)

**Comment by hgdsraj** [comment]
What changes are necessary?

**Comment by 50Wliu** [comment]
As I said in the review, this will need specs before it can be merged.

**Comment by thomasjo** [comment]
This change doesn't make sense. The original assignment of the `virtual` keyword was correct.
In other words, this is **not** the correct way to fix #170. Seems like that issue will need a deeper investigation, and it will likely require a slightly more tricky grammar change.

Such an effort should be started with a failing/red spec that tests the correct, expected behavior. Once the grammar bug has been resolved, that spec should become green.

Since it is likely less effort to start a fresh PR instead of rebasing this, and then starting over, I'm simply going to close this. If you want to help us out by fixing #170 properly, please open a new PR.  Feel free to open it early with the failing spec, and I'm sure someone will be able to help you figure out how to properly resolve this bug. Thanks! 🙇

**Pull request closed without merging**